### PR TITLE
option_menu to src

### PR DESCRIPTION
**Pull request by ghoulslash**
dissasemble asm/option_menu.s to src/option_menu.c

### CODE REVIEW

File: include/strings.h [hunk 1]
```diff
@@ -741,4 +741,28 @@ extern const u8 gText_DecimalPoint[];
...
+extern const u8 gText_BattleSceneOn[];
+extern const u8 gText_BattleSceneOff[];
+extern const u8 gText_BattleStyleShift[];
+extern const u8 gText_BattleStyleSet[];
+extern const u8 gText_SoundMono[];
+extern const u8 gText_SoundStereo[];
+extern const u8 gText_ButtonTypeNormal[];
+extern const u8 gText_ButtonTypeLR[];
+extern const u8 gText_ButtonTypeLEqualsA[];
+extern const u8 gText_MenuOptionOption[];
```
Comment (jiangzhengwenjz) [inline include/strings.h]: What does this mean?

File: data/strings.s [hunk 2]
```diff
@@ -2904,73 +2904,73 @@ gUnknown_8419D89:: @ 8419D89
...
+gText_ButtonTypeNormal:: @ 0x8419E46
 	.string "HELP$"
 
-gUnknown_8419E4B:: @ 0x8419E4B
+gText_ButtonTypeLR:: @ 0x8419E4B
 	.string "LゅA$"
 
-gUnknown_8419E4F:: @ 0x8419E4F
+gText_ButtonTypeLEqualsA:: @ 0x8419E4F
 	.string "LR$"
```
Comment (jiangzhengwenjz) [inline data/strings.s]: ?

File: src/option_menu.c [hunk 3]
```diff
@@ -0,0 +1,929 @@
...
+    MENUITEM_COUNT,
+};
+// Window Ids
+enum
+{
+    WIN_TEXT_OPTION,
+    WIN_OPTIONS,
+};
+
+static const struct WindowTemplate sOptionMenuWinTemplates[] =    //3CC2B8
```
Comment (jiangzhengwenjz) [inline src/option_menu.c]: you don't need to add comments for offset in ROM, unless it helps people figure out NONMATCHING

File: src/option_menu.c [hunk 4]
```diff
@@ -0,0 +1,929 @@
...
+static const u8 *const sSoundOptions[] =         //3CC34C
+{
+    gText_SoundMono, 
+    gText_SoundStereo
+};
+static const u8 *const sButtonTypeOptions[] =     //3CC354
+{
+    gText_ButtonTypeNormal, 
+    gText_ButtonTypeLEqualsA,
+    gText_ButtonTypeLR,
```
Comment (jiangzhengwenjz) [inline src/option_menu.c]: You don't have trailing comma in other arrays

File: src/option_menu.c [hunk 5]
```diff
@@ -0,0 +1,929 @@
...
+    MENUITEM_TEXTSPEED = 0,
+    MENUITEM_BATTLESCENE,
+    MENUITEM_BATTLESTYLE,
+    MENUITEM_SOUND,
+    MENUITEM_BUTTONMODE,
+    MENUITEM_FRAMETYPE,
+    MENUITEM_CANCEL,
+    MENUITEM_COUNT,
+};
+// Window Ids
```
Comment (jiangzhengwenjz) [inline src/option_menu.c]: newline

File: src/option_menu.c [hunk 6]
```diff
@@ -0,0 +1,929 @@
...
+        .tilemapLeft = 0,
+        .tilemapTop = 0,
+        .width = 30,
+        .height = 2,
+        .paletteNum = 0xF,
+        .baseBlock = 0x16e,
+    },
+    DUMMY_WIN_TEMPLATE,
+};
+static const struct BgTemplate sOptionMenuBgTemplates[] =    //3CC2D8
```
Comment (jiangzhengwenjz) [inline src/option_menu.c]: you also need newline here (this applies to everything below)

File: src/option_menu.c [hunk 7]
```diff
@@ -0,0 +1,929 @@
...
+{
+    /*0x00*/ u16 option[MENUITEM_COUNT];    //0,2,4,6,8,a,c
+    /*0x0E*/ u16 unkE;
+    /*0x10*/ u8 state3;
+    /*0x11*/ u8 state;
+    /*0x12*/ u8 state2;
+    /*0x13*/ u8 unk13;
+};
+
+EWRAM_DATA struct OptionsMenu *sOptionsMenuPtr = {0};
```
Comment (jiangzhengwenjz) [inline src/option_menu.c]: `static EWRAM_DATA struct OptionsMenu *sOptionsMenuPtr = NULL`

File: src/option_menu.c [hunk 8]
```diff
@@ -0,0 +1,929 @@
...
+
+
```
Comment (jiangzhengwenjz) [inline src/option_menu.c]: too many newlines

File: src/option_menu.c [hunk 9]
```diff
@@ -0,0 +1,929 @@
...
+
+//CB2_InitOptionsMenu
+static void sub_808835C(void)
```
Comment (jiangzhengwenjz) [inline src/option_menu.c]: can you rename the functions?

File: src/option_menu.c [hunk 10]
```diff
@@ -0,0 +1,929 @@
...
+//VBlankCB_OptionsMenu
+static void sub_8088374(void)
+{
+    LoadOam();
+    ProcessSpriteCopyRequests();
+    TransferPlttBuffer();
+}
+
+
+//sub_8088388
```
Comment (jiangzhengwenjz) [inline src/option_menu.c]: again we don't need this kind of comment (please also update below)

File: src/option_menu.c [hunk 11]
```diff
@@ -0,0 +1,929 @@
...
+        sub_8088430();
+        break;
+    case 1:
+        sub_8088530();
+        break;
+    case 2:
+        sub_80886D4();
+        break;
+    case 3:
+        if (sub_80886F0() != TRUE)    //silly matching quirk
```
Comment (jiangzhengwenjz) [inline src/option_menu.c]: This is actually common through the project, so the comment should be removed here

File: src/option_menu.c [hunk 12]
```diff
@@ -0,0 +1,929 @@
...
+static void sub_8088530(void)
+{
+    void * dest = (void *) VRAM;
+    DmaClearLarge16(3, dest, VRAM_SIZE, 0x1000);
+    
+    DmaClear32(3, (void *)OAM, OAM_SIZE);
+    DmaClear16(3, (void *)PLTT, PLTT_SIZE);    
+    SetGpuReg(REG_OFFSET_DISPCNT, 0);
+    ResetBgsAndClearDma3BusyFlags(0);
+    InitBgsFromTemplates(0, sOptionMenuBgTemplates, ARRAY_COUNT(sOptionMenuBgTemplates));
```
Comment (jiangzhengwenjz) [inline src/option_menu.c]: use `NELEMS` instead of `ARRAY_COUNT`

File: src/option_menu.c [hunk 13]
```diff
@@ -0,0 +1,929 @@
...
+    ChangeBgY(0, 0, 0);
+    ChangeBgX(1, 0, 0);
+    ChangeBgY(1, 0, 0);
+    ChangeBgX(2, 0, 0);
+    ChangeBgY(2, 0, 0);
+    ChangeBgX(3, 0, 0);
+    ChangeBgY(3, 0, 0);
+    InitWindows(sOptionMenuWinTemplates);
+    DeactivateAllTextPrinters();
+    SetGpuReg(REG_OFFSET_BLDCNT, 0xC1);
```
Comment (jiangzhengwenjz) [inline src/option_menu.c]: check `include/gba/io_reg.h` for available macros to replace `0xC1`

File: src/option_menu.c [hunk 14]
```diff
@@ -0,0 +1,929 @@
...
+    default:
+        return TRUE;
+    }
+    sOptionsMenuPtr->state2++;
+    return FALSE;
+}
+
+
+
+// I could not get this function to match. GOTO statements weren't even compiling correctly.
```
Comment (jiangzhengwenjz) [inline src/option_menu.c]: agbcc can tail merge cases. Try to write the code after goto respectively in those affected cases

File: src/option_menu.c [hunk 15]
```diff
@@ -0,0 +1,929 @@
...
+    ShowBg(1);
+    ShowBg(2);
+};
+
+//sub_8088680
+static void sub_8088680(void)
+{
+    s32 x;
+    x = 0xE4 - GetStringWidth(0, gText_PickSwitchCancel, 0);
+    FillWindowPixelBuffer(2, 0xFF);
```
Comment (jiangzhengwenjz) [inline src/option_menu.c]: use `PIXEL_FILL`macro

File: src/option_menu.c [hunk 16]
```diff
@@ -0,0 +1,929 @@
...
+    ChangeBgY(1, 0, 0);
+    ChangeBgX(2, 0, 0);
+    ChangeBgY(2, 0, 0);
+    ChangeBgX(3, 0, 0);
+    ChangeBgY(3, 0, 0);
+    InitWindows(sOptionMenuWinTemplates);
+    DeactivateAllTextPrinters();
+    SetGpuReg(REG_OFFSET_BLDCNT, 0xC1);
+    SetGpuReg(REG_OFFSET_BLDY, 2);
+    SetGpuReg(REG_OFFSET_WININ, 1);
```
Comment (jiangzhengwenjz) [inline src/option_menu.c]: check include/gba/io_reg.h for available macros to replace 1

File: src/option_menu.c [hunk 17]
```diff
@@ -0,0 +1,929 @@
...
+    SetGpuReg(REG_OFFSET_WINOUT, 0x27);
```
Comment (jiangzhengwenjz) [inline src/option_menu.c]: check include/gba/io_reg.h for available macros to replace 0x27

File: src/option_menu.c [hunk 18]
```diff
@@ -0,0 +1,929 @@
...
+    SetGpuReg(0, 0x3040);
```
Comment (jiangzhengwenjz) [inline src/option_menu.c]: check include/gba/io_reg.h for available macros to replace 0 and 0x3040

File: src/option_menu.c [hunk 19]
```diff
@@ -0,0 +1,929 @@
...
+                "\tbx r0\n");
+}
+#endif   
+
+//OptionsMenu_ProcessInput
+static u8 sub_80888C0(void)
+{ 
+    u16 current;
+    u16* curr;
+    if (gMain.newAndRepeatedKeys & DPAD_RIGHT)
```
Comment (jiangzhengwenjz) [inline src/option_menu.c]: use `JOY_REPT`macro

File: src/option_menu.c [hunk 20]
```diff
@@ -0,0 +1,929 @@
...
+        if (current == (sOptionsMenuItemCounts[sOptionsMenuPtr->unkE] - 1))
+            sOptionsMenuPtr->option[sOptionsMenuPtr->unkE] = 0;
+        else
+            sOptionsMenuPtr->option[sOptionsMenuPtr->unkE] = current + 1;
+        if (sOptionsMenuPtr->unkE == MENUITEM_FRAMETYPE)
+            return 2;
+        else
+            return 4;
+    }
+    else if (gMain.newAndRepeatedKeys & DPAD_LEFT)
```
Comment (jiangzhengwenjz) [inline src/option_menu.c]: use `JOY_REPT` macro

File: src/option_menu.c [hunk 21]
```diff
@@ -0,0 +1,929 @@
...
+    {
+        curr = &sOptionsMenuPtr->option[sOptionsMenuPtr->unkE];
+        if (*curr == 0)
+            *curr = sOptionsMenuItemCounts[sOptionsMenuPtr->unkE] - 1;
+        else
+            --*(curr);
```
Comment (jiangzhengwenjz) [inline src/option_menu.c]: redundant parentheses

File: src/option_menu.c [hunk 22]
```diff
@@ -0,0 +1,929 @@
...
+        
+        if (sOptionsMenuPtr->unkE == MENUITEM_FRAMETYPE)
+            return 2;
+        else
+            return 4;
+    }
+    else if (gMain.newAndRepeatedKeys & DPAD_UP)
```
Comment (jiangzhengwenjz) [inline src/option_menu.c]: use `JOY_REPT` macro

File: src/option_menu.c [hunk 23]
```diff
@@ -0,0 +1,929 @@
...
+    {
+        if (sOptionsMenuPtr->unkE == MENUITEM_TEXTSPEED)
+            sOptionsMenuPtr->unkE = MENUITEM_CANCEL;
+        else
+            sOptionsMenuPtr->unkE = sOptionsMenuPtr->unkE - 1;
+        return 3;        
+    }
+    else if (gMain.newAndRepeatedKeys & DPAD_DOWN)
```
Comment (jiangzhengwenjz) [inline src/option_menu.c]: use `JOY_REPT` macro

File: src/option_menu.c [hunk 24]
```diff
@@ -0,0 +1,929 @@
...
+    {
+        if (sOptionsMenuPtr->unkE == MENUITEM_CANCEL)
+            sOptionsMenuPtr->unkE = MENUITEM_TEXTSPEED;
+        else
+            sOptionsMenuPtr->unkE = sOptionsMenuPtr->unkE + 1;
+        return 3;
+    }
+    else if ((gMain.newKeys & B_BUTTON) || (gMain.newKeys & A_BUTTON))
```
Comment (jiangzhengwenjz) [inline src/option_menu.c]: use `JOY_NEW` macro

File: src/option_menu.c [hunk 25]
```diff
@@ -0,0 +1,929 @@
...
+    }
+    else
+    {
+        return 0;
+    }
+}
+
+
+//sub_80889A8
+#ifdef NONMATCHING    // could not get it to match perfectly, no idea how they put so many vars on the stack
```
Comment (jiangzhengwenjz) [inline src/option_menu.c]: it may be local arrays. can you setup compiler explorer for this?

File: src/option_menu.c [hunk 26]
```diff
@@ -0,0 +1,929 @@
...
+        break;
+    case MENUITEM_SOUND:
+        AddTextPrinterParameterized3(1, 2, x, y, dst, -1, sSoundOptions);
+        break;
+    case MENUITEM_BUTTONMODE:
+        AddTextPrinterParameterized3(1, 2, x, y, dst, -1, sButtonTypeOptions);
+        break;
+    case MENUITEM_FRAMETYPE:
+        StringCopy(str, gText_FrameType);
+        ConvertIntToDecimalStringN(v8, sOptionsMenuPtr->option[2*selection] + 1, 1, 2);
```
Comment (jiangzhengwenjz) [inline src/option_menu.c]: space on left and right of `*`

File: src/option_menu.c [hunk 27]
```diff
@@ -0,0 +1,929 @@
...
+    
+    gSaveBlock2Ptr->optionsTextSpeed = (u8) sOptionsMenuPtr->option[MENUITEM_TEXTSPEED];
+    gSaveBlock2Ptr->optionsBattleSceneOff = (u8) sOptionsMenuPtr->option[MENUITEM_BATTLESCENE];
+    gSaveBlock2Ptr->optionsBattleStyle = (u8) sOptionsMenuPtr->option[MENUITEM_BATTLESTYLE];
+    gSaveBlock2Ptr->optionsSound = sOptionsMenuPtr->option[MENUITEM_SOUND];
+    gSaveBlock2Ptr->optionsButtonMode = (u8) sOptionsMenuPtr->option[MENUITEM_BUTTONMODE];
+    gSaveBlock2Ptr->optionsWindowFrameType = (u8) sOptionsMenuPtr->option[MENUITEM_FRAMETYPE];
+    SetPokemonCryStereo(gSaveBlock2Ptr->optionsSound);
+    Free(sOptionsMenuPtr);
+    sOptionsMenuPtr = NULL;
```
Comment (jiangzhengwenjz) [inline src/option_menu.c]: use `FREE_AND_SET_NULL` macro

File: src/option_menu.c [hunk 28]
```diff
@@ -0,0 +1,929 @@
...
+
+//sub_8088DE0
+static void sub_8088DE0(u16 selection)
+{
+    u16 v1, v2;
+    
+    v1 = GetFontAttribute(2, FONTATTR_MAX_LETTER_HEIGHT);
+    v2 = selection * (v1 - 1) + 0x3A;
+    SetGpuReg(0x44, (v2 << 8) | (v2 + v1));
+    SetGpuReg(0x40, 0x10E0);
```
Comment (jiangzhengwenjz) [inline src/option_menu.c]: again, check io_reg.h

File: src/option_menu.c [hunk 29]
```diff
@@ -0,0 +1,929 @@
...
+
+
+//sub_8088DE0
+static void sub_8088DE0(u16 selection)
+{
+    u16 v1, v2;
+    
+    v1 = GetFontAttribute(2, FONTATTR_MAX_LETTER_HEIGHT);
+    v2 = selection * (v1 - 1) + 0x3A;
+    SetGpuReg(0x44, (v2 << 8) | (v2 + v1));
```
Comment (jiangzhengwenjz) [inline src/option_menu.c]: again, check io_reg.h

File: src/option_menu.c [hunk 30]
```diff
@@ -0,0 +1,929 @@
...
+
+//CB2_InitOptionsMenu
+static void sub_808835C(void)
+{
+    RunTasks();
+    AnimateSprites();
+    BuildOamBuffer();
+    UpdatePaletteFade();
+}
+
```
Comment (jiangzhengwenjz) [inline src/option_menu.c]: redundant newline (please also check below)

File: src/option_menu.c [hunk 31]
```diff
@@ -0,0 +1,929 @@
...
+#ifdef NONMATCHING
+static void sub_8088780(u8 taskId)
+{
+    u8 v2, v5;
+    struct OptionsMenu v4;
+    
+    switch (sOptionsMenuPtr->state3)
+    {
+    case 0:
+        BeginNormalPaletteFade(-1, 0, 0x10, 0, 0);
```
Comment (jiangzhengwenjz) [inline src/option_menu.c]: -1 -> 0xFFFFFFFF, this is a u32 mask
also use `RGB_BLACK` macro

File: src/option_menu.c [hunk 32]
```diff
@@ -0,0 +1,929 @@
...
+            break;
+        case 4:
+            sub_80889A8(sOptionsMenuPtr->unkE);
+            sOptionsMenuPtr->state3++;
+            break;
+        default:
+            return;
+        }
+    case 3:
+        BeginNormalPaletteFade(-1, 0, 0, 0x10, 0);
```
Comment (jiangzhengwenjz) [inline src/option_menu.c]: 0xFFFFFFFF
also use `RGB_BLACK` macro

File: src/option_menu.c [hunk 33]
```diff
@@ -0,0 +1,929 @@
...
+    sOptionsMenuPtr = NULL;
+    DestroyTask(taskId);
+}
+
+    
+    
+//sub_8088BD0
+static void sub_8088BD0(void)
+{
+    FillWindowPixelBuffer(0, 0x11);
```
Comment (jiangzhengwenjz) [inline src/option_menu.c]: `PIXEL_FILL`

File: src/option_menu.c [hunk 34]
```diff
@@ -0,0 +1,929 @@
...
+    FillBgTilemapBufferRect(1, 0x1B2, 0x1C, 0x13, 1, 1, h);
+    CopyBgTilemapBufferToVram(1);
+}
+
+//sub_8088D8C
+static void sub_8088D8C(void)
+{
+    u8 i;
+    
+    FillWindowPixelBuffer(1, 0x11);
```
Comment (jiangzhengwenjz) [inline src/option_menu.c]: `PIXEL_FILL`

File: src/option_menu.c [hunk 35]
```diff
@@ -0,0 +1,929 @@
...
+    for (i = 0; i < MENUITEM_COUNT; i++)
+    {
+        AddTextPrinterParameterized(1, 2, sOptionMenuItemsNames[i], 8, (u8) ((i*(GetFontAttribute(2, FONTATTR_MAX_LETTER_HEIGHT))) + 2) - i, 0xFF, 0);
```
Comment (jiangzhengwenjz) [inline src/option_menu.c]: 1) space on left and right of `*`.
2) 0 -> NULL
3) why did you use TEXT_SPEED_FF above but not here?

File: src/option_menu.c [hunk 36]
```diff
@@ -0,0 +1,929 @@
...
+        return 0;
+    }
+}
+
+
+//sub_80889A8
+#ifdef NONMATCHING    // could not get it to match perfectly, no idea how they put so many vars on the stack
+static void sub_80889A8(u8 selection)
+{
+    u8* dst;
```
Comment (PikalaxALT) [inline src/option_menu.c]: `u8 dst[3];`

File: src/option_menu.c (hunk 28 above)
Comment (ghoulslash) [reply src/option_menu.c]: 0x40 is REG_OFFSET_WIN0H, but no corresponding macro for 0xC0

Comment (jiangzhengwenjz) [reply src/option_menu.c]: `WIN_RANGE`

File: src/option_menu.c [hunk 37]
```diff
@@ -0,0 +1,903 @@
...
+    sOptionMenuPtr->state = 0;
+    sOptionMenuPtr->unkE = 0;
+    sOptionMenuPtr->option[MENUITEM_TEXTSPEED] = gSaveBlock2Ptr->optionsTextSpeed;
+    sOptionMenuPtr->option[MENUITEM_BATTLESCENE] = gSaveBlock2Ptr->optionsBattleSceneOff;
+    sOptionMenuPtr->option[MENUITEM_BATTLESTYLE] = gSaveBlock2Ptr->optionsBattleStyle;
+    sOptionMenuPtr->option[MENUITEM_SOUND] = gSaveBlock2Ptr->optionsSound;
+    sOptionMenuPtr->option[MENUITEM_BUTTONMODE] = gSaveBlock2Ptr->optionsButtonMode;
+    sOptionMenuPtr->option[MENUITEM_FRAMETYPE] = gSaveBlock2Ptr->optionsWindowFrameType;
+    
+    for (i = 0; i < MENUITEM_COUNT-1; i++)
```
Comment (jiangzhengwenjz) [inline src/option_menu.c]: space at left and right of `-`

File: src/option_menu.c [hunk 38]
```diff
@@ -0,0 +1,903 @@
...
+    {
+        if (sOptionMenuPtr->option[i] > (sOptionMenuItemCounts[i])-1)
```
Comment (jiangzhengwenjz) [inline src/option_menu.c]: space at left and right of `-`

File: src/option_menu.c [hunk 39]
```diff
@@ -0,0 +1,903 @@
...
+            sOptionMenuPtr->option[i] = 0;
+    }
+    HelpSystem_SetSomeVariable2(0xD);
+    SetMainCallback2(CB2_OptionMenu);
+}
+
+static void OptionMenu_InitCallbacks(void)
+{
+    SetVBlankCallback(0);
```
Comment (jiangzhengwenjz) [inline src/option_menu.c]: `NULL`

File: src/option_menu.c [hunk 40]
```diff
@@ -0,0 +1,903 @@
...
+    SetHBlankCallback(0);
```
Comment (jiangzhengwenjz) [inline src/option_menu.c]: `NULL`

File: src/option_menu.c [hunk 41]
```diff
@@ -0,0 +1,903 @@
...
+            BufferOptionMenuString(i);
+        break;
+    case 8:
+        sub_8088DE0(sOptionMenuPtr->unkE);
+        break;
+    case 9:
+        OptionMenu_PickSwitchCancel();
+        break;
+    default:
+        SetOptionMenuTask();
```
Comment (jiangzhengwenjz) [inline src/option_menu.c]: add a `break` here (check style guide)

File: src/option_menu.c [hunk 42]
```diff
@@ -0,0 +1,903 @@
...
+    }
+}
+
+static void sub_8088DE0(u16 selection)
+{
+    u16 v1, v2;
+    
+    v1 = GetFontAttribute(2, FONTATTR_MAX_LETTER_HEIGHT);
+    v2 = selection * (v1 - 1) + 0x3A;
+    SetGpuReg(REG_OFFSET_WIN0V, (v2 << 8) | (v2 + v1));
```
Comment (jiangzhengwenjz) [inline src/option_menu.c]: use `WIN_RANGE` macro

File: src/option_menu.c [hunk 43]
```diff
@@ -0,0 +1,903 @@
...
+    SetGpuReg(REG_OFFSET_WIN0H, WINOUT_WINOBJ_OBJ | WIN_RANGE(0, 0xE0));
```
Comment (jiangzhengwenjz) [inline src/option_menu.c]: `WINOUT_WINOBJ_OBJ | WIN_RANGE(0, 0xE0)` -> `WIN_RANGE(0x10, 0xE0)`

File: src/option_menu.c [hunk 44]
```diff
@@ -0,0 +1,903 @@
...
+{
+    /*0x00*/ u16 option[MENUITEM_COUNT];
+    /*0x0E*/ u16 unkE;
+    /*0x10*/ u8 state3;
+    /*0x11*/ u8 state;
+    /*0x12*/ u8 state2;
+    /*0x13*/ u8 unk13;
+};
+
+EWRAM_DATA struct OptionMenu *sOptionMenuPtr = NULL;
```
Comment (jiangzhengwenjz) [inline src/option_menu.c]: `static` is still missing

File: data/strings.s [hunk 45]
```diff
@@ -2955,16 +2955,16 @@ gText_SoundMono:: @ 0x8419E36
...
-gUnknown_8419E42:: @ 0x8419E42
+gTextJP_Type:: @ 0x8419E42
 	.string "タイプ$"
 
 gText_ButtonTypeNormal:: @ 0x8419E46
 	.string "HELP$"
 
-gText_ButtonTypeLR:: @ 0x8419E4B
+gText_ButtonTypeLEqualsA:: @ 0x8419E4B
 	.string "LゅA$"
```
Comment (PikalaxALT) [inline data/strings.s]: Fix the Kana here (use charmap.txt to identify the equivalent codepoint)

File: data/strings.s [hunk 46]
```diff
@@ -2955,16 +2955,16 @@ gText_SoundMono:: @ 0x8419E36
 gText_SoundStereo:: @ 0x8419E3B
 	.string "STEREO$"
 
-gUnknown_8419E42:: @ 0x8419E42
+gTextJP_Type:: @ 0x8419E42
 	.string "タイプ$"
 
 gText_ButtonTypeNormal:: @ 0x8419E46
```
Comment (PikalaxALT) [inline data/strings.s]: gText_ButtonTypeHelp

File: src/option_menu.c [hunk 47]
```diff
@@ -30,16 +29,51 @@ enum
...
+
 // Window Ids
 enum
 {
     WIN_TEXT_OPTION,
-    WIN_OPTIONS,
+    WIN_OPTIONS
+};
+
+// RAM sumbols
```
Comment (PikalaxALT) [inline src/option_menu.c]: Is "sumbols" a common English word?

File: src/item_pc.c [hunk 48]
```diff
@@ -384,7 +384,7 @@ static bool8 ItemPc_DoGfxSetup(void)
         gMain.state++;
         break;
     case 20:
-        if (sub_80BF72C() != TRUE)
+        if ((u8) sub_80BF72C() != TRUE)
```
Comment (PikalaxALT) [inline src/item_pc.c]: No space after the cast closing parenthesis

File: src/option_menu.c [hunk 49]
```diff
@@ -0,0 +1,579 @@
...
+        break;
+    default:
+        return TRUE;
+    }
+    sOptionMenuPtr->state2++;
+    return FALSE;
+}
+
+
+//#ifdef NONMATCHING
```
Comment (PikalaxALT) [inline src/option_menu.c]: Remove comment

File: src/option_menu.c [hunk 50]
```diff
@@ -0,0 +1,579 @@
...
+        sub_814FDA0(1, 0x1B3, 0x30);
+        break;
+    default:
+        return TRUE;
+    }
+    sOptionMenuPtr->state2++;
+    return FALSE;
+}
+
+
```
Comment (PikalaxALT) [inline src/option_menu.c]: Excess newline

File: src/option_menu.c [hunk 51]
```diff
@@ -0,0 +1,579 @@
...
+static void SetOptionMenuTask(void)
+{
+    CreateTask(Task_OptionMenu, 0);
+    SetMainCallback2(CB2_InitOptionMenu);
+}
+
+static void InitOptionMenuBg(void)
+{
+    void * dest = (void *) VRAM;
```
Comment (PikalaxALT) [inline src/option_menu.c]: Extra space after cast closing parenthesis

File: src/option_menu.c [hunk 52]
```diff
@@ -514,12 +512,12 @@ static void CloseAndSaveOptionMenu(u8 taskId)
     gFieldCallback = sub_807DF64;
     SetMainCallback2(gMain.savedCallback);
     FreeAllWindowBuffers();
-    gSaveBlock2Ptr->optionsTextSpeed = (u8) sOptionMenuPtr->option[MENUITEM_TEXTSPEED];
-    gSaveBlock2Ptr->optionsBattleSceneOff = (u8) sOptionMenuPtr->option[MENUITEM_BATTLESCENE];
-    gSaveBlock2Ptr->optionsBattleStyle = (u8) sOptionMenuPtr->option[MENUITEM_BATTLESTYLE];
+    gSaveBlock2Ptr->optionsTextSpeed = (u8)sOptionMenuPtr->option[MENUITEM_TEXTSPEED];
```
Comment (PikalaxALT) [inline src/option_menu.c]: Are these casts necessary to match?

**Pull request merged**